blueprints/route/index.js

```diff
@@ -29,6 +29,11 @@ module.exports = {
       name: 'reset-namespace',
       type: Boolean,
     },
+    {
+      name: 'strict-template',
```

Comment:
Not sure if this is the best name for the option. Suggestions?

Reply:
Another option could be to name it something like `--strict-route-template` but also add a `--strict` alias. That way people can add the more verbose version to their ember-cli file in (the unlikely) case they don't want to use the strict option for the components version.

Reply:
A third option is to match the setup from the [component PR](https://github.com/emberjs/ember.js/pull/20862/files#diff-b3654ccb6dc2075bf987b07346f57bfd89f7874ca631cf76040a1afe49ee0a9eR53).

And add a `route-template-authoring-format` option with `--loose` and `--strict` shorthands.

Reply:
We went with `route-authoring-format` since that's the most consistent with the component version.